lib/nrf_modem_lib/nrf_modem_lib_trace_sync.c

```diff
@@ -107,32 +52,7 @@ int nrf_modem_lib_trace_process(const uint8_t *data, uint32_t len)
 		return 0;
 	}
 
-#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_UART
-	/* Split RAM buffer into smaller chunks to be transferred using DMA. */
-	uint32_t remaining_bytes = len;
-	const uint32_t MAX_BUF_LEN = (1 << UARTE1_EASYDMA_MAXCNT_SIZE) - 1;
-
-	while (remaining_bytes) {
-		size_t transfer_len = MIN(remaining_bytes, MAX_BUF_LEN);
-		uint32_t idx = len - remaining_bytes;
-
-		nrfx_uarte_tx(&uarte_inst, &data[idx], transfer_len);
-		remaining_bytes -= transfer_len;
-	}
-#endif
-
-#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT
-	uint32_t remaining_bytes = len;
-
-	while (remaining_bytes) {
-		uint8_t transfer_len = MIN(remaining_bytes,
-					CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT_BUF_SIZE);
-		uint32_t idx = len - remaining_bytes;
-
-		SEGGER_RTT_WriteSkipNoLock(trace_rtt_channel, &data[idx], transfer_len);
-		remaining_bytes -= transfer_len;
-	}
-#endif
+	trace_medium_write(data, len);
```

Comment:
Aren't we missing the trace processed callback?

Reply:
With `CONFIG_NRF_MODEM_LIB_TRACE_THREAD_PROCESSING=n`, the trace processed callback is in `nrf_modem_os_trace_put()`. Will try to move it into `nrf_modem_lib_trace_sync.c` to get rid the preprocessor-if in `nrf_modem_os_trace_put()`.